test/kb_fasttree_server_test.py

```diff
@@ -139,15 +182,49 @@ def test_kb_fasttree_run_FastTree_01(self):
             "gamma": "0",
         }
 
-        ret = self.getImpl().run_FastTree(self.getContext(), parameters)[0]
-        self.assertIsNotNone(ret["report_ref"])
-
-        # check created obj
-        report_obj = self.getWsClient().get_objects([{"ref": ret["report_ref"]}])[0]["data"]
-        self.assertIsNotNone(report_obj["objects_created"][0]["ref"])
-
-        created_obj_0_info = self.getWsClient().get_object_info_new(
-            {"objects": [{"ref": report_obj["objects_created"][0]["ref"]}]}
-        )[0]
-        self.assertEqual(created_obj_0_info[NAME_I], obj_out_name)
-        self.assertEqual(created_obj_0_info[TYPE_I].split("-")[0], obj_out_type)
+        # patch generate_run_id to return self.run_id
+        with patch("kb_fasttree.fasttree.generate_run_id", return_value=self.run_id):
+            ret = self.get_impl().run_FastTree(self.get_context(), parameters)[0]
+            assert ret["report_ref"] is not None
+
+            # check created obj
+            report_obj = self.get_ws_client().get_objects([{"ref": ret["report_ref"]}])[0]["data"]
+            assert report_obj["objects_created"][0]["ref"] is not None
+
+            created_obj_0_info = self.get_ws_client().get_object_info_new(
+                {"objects": [{"ref": report_obj["objects_created"][0]["ref"]}]}
+            )[0]
+            assert created_obj_0_info[1] == obj_out_name
+            assert created_obj_0_info[2].split("-")[0] == obj_out_type
+
+            # check that the expected files are in the expected places
+            run_dir = Path(self.cfg["scratch"]) / self.run_id
+            assert run_dir.exists()
+
+            for sub_dir_name in ["output", "output_html"]:
+                sub_dir = run_dir / sub_dir_name
+                assert sub_dir.exists()
+
+            check_non_zero_file_size(run_dir / f"{MSA_NAME}.fasta")
+            compare_files(
+                run_dir / f"{MSA_NAME}.fasta",
+                TEST_BASE_DIR / "data" / "expected" / f"{MSA_NAME}.fasta",
+            )
+
+            file_list = {
+                "output": [
+                    f"{obj_out_name}{suffix}" for suffix in ["-labels.newick", ".newick", ".pdf"]
+                ],
+                "output_html": [
+                    f"{obj_out_name}.html",
+                    f"{obj_out_name}.png",
+                ],  # , "output_html.zip"],
+            }
+            for sub_dir_name in file_list:
+                for f in file_list[sub_dir_name]:
+                    check_non_zero_file_size(run_dir / sub_dir_name / f)
+                    if not f.endswith("pdf") and not f.endswith("png"):
```

Comment:
don't check binary files as the `compare_files` function only looks at text files